packages/p2p-v2/src/routes/AppContent/index.tsx

```diff
@@ -14,16 +15,29 @@ export const routesConfiguration = [
     { Component: <div> My Ads Page </div>, path: 'my-ads', title: 'My Ads' },
     { Component: <MyProfile />, path: 'my-profile', title: 'My Profile' },
 ];
+
+const pathToTitleMapper = Object.fromEntries(routesConfiguration.map(route => [route.path, route.title]));
+
+const getCurrentRoute = () => {
+    const segments = new URL(window.location.href).pathname.split('/');
+    const endPath = segments.pop();
+    return endPath;
+};
+
 const AppContent = () => {
     const history = useHistory();
     const { data: activeAccountData, isLoading } = useActiveAccount();
+    const [activeTab, setActiveTab] = useState(() => pathToTitleMapper[getCurrentRoute() || DEFAULT_TAB]);
 
-    const initialTab = useMemo(() => {
-        const pathname = new URL(window.location.href).pathname;
-        const segments = pathname.split('/');
+    useEventListener('switchTab', event => {
+        setActiveTab(pathToTitleMapper[event.detail.tab]);
+        history.push(`/cashier/p2p-v2/${event.detail.tab}`);
+    });
 
-        return routesConfiguration.find(route => route.path === segments[segments.length - 1])?.title || DEFAULT_TAB;
-    }, [window.location.href]);
+    useEventListener('popstate', () => {
+        const endPath = getCurrentRoute();
+        if (endPath) setActiveTab(pathToTitleMapper[endPath]);
+    });
```

Comment:
This fixes an issue where routing to a different tab is not switched, as well as the tabs not tracking the history stack:
https://github.com/binary-com/deriv-app/assets/103016120/69d6c203-3c7e-4a8d-8cc2-1166bb971fa8